service/src/main/java/bio/terra/workspace/app/controller/ControlledGcpResourceApiController.java

```diff
@@ -988,12 +1003,43 @@ public ResponseEntity<ApiGcpDataprocClusterResource> updateDataprocCluster(
             .setName(requestBody.getName())
             .setDescription(requestBody.getDescription());
 
-    // TODO: PF-2901 Add update parameters when update is supported
+    // Validate update parameter values to ensure there are no invalid configurations that can cause
+    // dismal failures.
+    ApiControlledDataprocClusterUpdateParameters updateParameters =
+        requestBody.getUpdateParameters();
+    if (updateParameters != null) {
+      if (updateParameters.getLifecycleConfig() != null) {
+        // Cluster scheduled deletion configurations cannot be updated in tandem with other
+        // attribute
+        // updates.
+        if (updateParameters.getNumPrimaryWorkers() != null
+            || updateParameters.getNumSecondaryWorkers() != null
+            || updateParameters.getAutoscalingPolicy() != null) {
+          throw new BadRequestException(
+              "Cluster scheduled deletion configurations cannot be updated in tandem with other attribute updates.");
+        }
+        // Can only specify one of idleDeleteTtl or autoDeleteTtl or autoDeleteTime
```

Comment:
Why only one here? When creating, it seems like you can specify both `idleDeleteTtl` and (`autoDeleteTtl` or `autoDeleteTime`)

Reply:
You're right, I've updated the validation.